include/CL/opencl.hpp

```diff
@@ -548,19 +548,19 @@
 
 // Define deprecated prefixes and suffixes to ensure compilation
 // in case they are not pre-defined
-#if !defined(CL_EXT_PREFIX__VERSION_1_1_DEPRECATED)
-#define CL_EXT_PREFIX__VERSION_1_1_DEPRECATED  
-#endif // #if !defined(CL_EXT_PREFIX__VERSION_1_1_DEPRECATED)
-#if !defined(CL_EXT_SUFFIX__VERSION_1_1_DEPRECATED)
-#define CL_EXT_SUFFIX__VERSION_1_1_DEPRECATED
-#endif // #if !defined(CL_EXT_PREFIX__VERSION_1_1_DEPRECATED)
-
-#if !defined(CL_EXT_PREFIX__VERSION_1_2_DEPRECATED)
-#define CL_EXT_PREFIX__VERSION_1_2_DEPRECATED  
-#endif // #if !defined(CL_EXT_PREFIX__VERSION_1_2_DEPRECATED)
-#if !defined(CL_EXT_SUFFIX__VERSION_1_2_DEPRECATED)
-#define CL_EXT_SUFFIX__VERSION_1_2_DEPRECATED
-#endif // #if !defined(CL_EXT_PREFIX__VERSION_1_2_DEPRECATED)
+#if !defined(CL_API_PREFIX__VERSION_1_1_DEPRECATED)
+#define CL_API_PREFIX__VERSION_1_1_DEPRECATED  
+#endif // #if !defined(CL_API_PREFIX__VERSION_1_1_DEPRECATED)
+#if !defined(CL_API_SUFFIX__VERSION_1_1_DEPRECATED)
+#define CL_API_SUFFIX__VERSION_1_1_DEPRECATED
+#endif // #if !defined(CL_API_PREFIX__VERSION_1_1_DEPRECATED)
+
+#if !defined(CL_API_PREFIX__VERSION_1_2_DEPRECATED)
+#define CL_API_PREFIX__VERSION_1_2_DEPRECATED  
+#endif // #if !defined(CL_API_PREFIX__VERSION_1_2_DEPRECATED)
+#if !defined(CL_API_SUFFIX__VERSION_1_2_DEPRECATED)
+#define CL_API_SUFFIX__VERSION_1_2_DEPRECATED
+#endif // #if !defined(CL_API_PREFIX__VERSION_1_2_DEPRECATED)
```

Comment:
Could we add the `VERSION_2_2_DEPRECATED` prefixes and suffixes here in case they are not pre-defined?  This would enable CI to pass even without the header updates, which is good because it's very possible that the the headers and the C++ bindings could be out-of-sync for a bit.

Reply:
I'm remembering now that I added the 2.2 deprecation guards (https://github.com/KhronosGroup/OpenCL-CLHPP/pull/118).  I should have added the "in case they are not pre-defined" 2.2 prefixes and suffixes here at the same time - sorry about that!

This raises an interesting question though: should the functions in the C++ bindings _also_ include the deprecation prefixes and suffixes?  Or, is this overkill, since the OpenCL API calls will already include a deprecation message?

Reply:
More to the point.  In the C++ bindings we have (this is a nice short example, there are other places that use the same pattern):

```c++
#if defined(CL_USE_DEPRECATED_OPENCL_1_1_APIS)
/**
 * Unload the OpenCL compiler.
 * \note Deprecated for OpenCL 1.2. Use Platform::unloadCompiler instead.
 */
inline CL_API_PREFIX__VERSION_1_1_DEPRECATED cl_int
UnloadCompiler() CL_API_SUFFIX__VERSION_1_1_DEPRECATED;
inline cl_int
UnloadCompiler()
{
    return ::clUnloadCompiler();
}
#endif // #if defined(CL_USE_DEPRECATED_OPENCL_1_1_APIS)
```

When we have `CL_USE_DEPRECATED_OPENCL_1_1_APIS` defined, `CL_API_PREFIX__VERSION_1_1_DEPRECATED` is always defined to nothing, so it's not adding anything.  Even if we didn't have the guard for `CL_USE_DEPRECATED_OPENCL_1_1_APIS`, we will already get a warning that `clUnloadCompiler` was declared deprecated, so do we really need a deprecation warning on `UnloadCompiler` also?